shared/src/models/utils/vehicle-resource.ts

```diff
@@ -0,0 +1,22 @@
+import { ValidateNested } from 'class-validator';
+import { IsValue } from '../../utils/validators';
+import { IsResourceDescription } from '../../utils/validators/is-resource-description';
+import { getCreate } from './get-create';
+
+export class VehicleResource {
+    @IsValue('vehicleResource' as const)
+    public readonly type = 'vehicleResource';
+
+    @ValidateNested()
+    @IsResourceDescription()
+    public readonly vehicleTypes!: { [key: string]: number };
```

Comment:
I suggest naming this `vehicleCount`, as it maps vehicle types to their vehicle count